src/sysutil.rs

```diff
@@ -191,12 +193,65 @@ impl SysUtil {
                     panic!("Session file '{}' is not UTF-8: {}", path.display(), err)
                 });
 
+                let stem = if let Some(stem) = path.file_stem() {
+                    // Get the stem of the filename of this desktop file.
+                    if let Some(stem) = stem.to_str() {
+                        stem.to_string()
+                    } else {
+                        warn!("Non-UTF-8 file stem in session file: {}", path.display());
+                        // No way to display this session name, so just skip it.
+                        continue;
+                    }
+                } else {
+                    warn!("No file stem found for session: {}", path.display());
+                    // No file stem implies no file name, which shouldn't happen.
+                    // Since there's no full name nor file stem, just skip this anomalous
+                    // session.
+                    continue;
+                };
+                if stems.contains(&stem) {
+                    debug!(
+                        "{stem} was already found elsewhere, skipping {}",
+                        path.display()
+                    );
+                    continue;
+                };
+
+                stems.insert(stem.to_string());
```

Comment:
Shouldn't `stem` already be a string? So is the `.to_string()` needed?

Reply:
Also, let's move this insertion to the end of the loop, right after we insert this session into `sessions`, so that we don't have to run `stems.remove(&stem)` whenever we want to `continue`.

Reply:
> Shouldn't `stem` already be a string? So is the `.to_string()` needed?

I'm honestly not sure - as I said, first time writing anything in Rust. I just know, that without this, compiler was throwing errors at me

Reply:
> Also, let's move this insertion to the end of the loop, right after we insert this session into `sessions`, so that we don't have to run `stems.remove(&stem)` whenever we want to `continue`.

I'm fine with that, however then I will have to include it in the moment I'm skipping an entry due to `Hidden/NoDisplay=true`

Reply:
> I'm honestly not sure - as I said, first time writing anything in Rust. I just know, that without this, compiler was throwing errors at me

Then let's keep it that way to respect the compiler 😄 

> I'm fine with that, however then I will have to include it in the moment I'm skipping an entry due to Hidden/NoDisplay=true

Ah right, then it's up to you where to add it.